ReactAndroid/src/main/java/com/facebook/react/common/TestIdUtil.java

```diff
@@ -0,0 +1,69 @@
+package com.facebook.react.common;
+
+import android.view.View;
+
+import com.facebook.react.common.annotations.VisibleForTesting;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Utility methods for managing testIDs on views and mapping them back to React Tags.
+ */
+public class TestIdUtil {
+    private static final ConcurrentHashMap<String, Integer> TEST_IDS = new ConcurrentHashMap<>();
+    private static final ConcurrentHashMap<Integer, Integer> ORIGINAL_REACT_TAGS = new ConcurrentHashMap<>();
```

Comment:
Will these two maps just keep growing during the lifetime of the app? If React components are mounted in the app, will entries be added here? But if components are unmounted, will they not be removed? If that's the way this works, that's not good for memory consumption.